src/Workspaces/CSharp/Portable/CodeGeneration/ParameterGenerator.cs

```diff
@@ -152,6 +152,7 @@ internal static ParameterSyntax GetParameter(IParameterSymbol p, CodeGenerationO
             {
                 return default;
             }
+            attributes = attributes.WhereAsArray(a => a.ToString() != "System.Runtime.CompilerServices.IsReadOnlyAttribute");
```

Comment:
😨 A start would be `WellKnownTypes.GetMetadataName(WellKnownType.System_Runtime_CompilerServices_IsReadOnlyAttribute)`

❓ Why not move this to `AttributeGenerator.GenerateAttribute`? Returning null from that method suppresses the syntax.

💭 It makes me wonder if this should be handled more like `System.Runtime.CompilerServices.ExtensionAttribute`.

Reply:
The reason I didn't move this filtering into `GenerateAttribute` is that it should only apply when the attribute is applied on a parameter. If it was applied somewhere else (for some reason), it should be printed, I think.

Reply:
`WellKnownTypes` is inaccessible.

I looked around to see how `ExtensionAttribute` is handled, but couldn't find any relevant logic. Since extensions methods are static, they can't be overridden, so it's possible that the case is just not handled.


---
In reply to: [164312453](https://github.com/dotnet/roslyn/pull/24492#discussion_r164312453) [](ancestors = 164312453)

Reply:
💭 Since we already have `IParameterSymbol.RefKind`, it seems ideal to remove the attribute from the list of attributes provided by the API.